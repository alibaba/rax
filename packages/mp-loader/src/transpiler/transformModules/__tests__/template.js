const { join } = require('path');
const { transformNode, genData } = require('../template');

describe('Transpile module: template', () => {
  it('import', () => {
    const src = './__files__/template.axml';
    const el = {
      tag: 'import',
      attrsMap: { src },
      attrsList: [{
        name: 'src', value: src,
      }]
    };
    transformNode(el, { templatePath: __filename });
    expect(el.isComment).toEqual(true);
    expect(el.attrsList).toEqual([]);
    expect(el.type).toEqual(3);
    expect(el.templates).toEqual([join(__dirname, src)]);
    expect(genData(el)).toMatchSnapshot();
  });

  it('template register', () => {
    const name = 'foo';
    const el = {
      tag: 'template',
      attrsMap: { name },
      attrsList: [{
        name: 'name', value: name,
      }]
    };
    transformNode(el, {});
    expect(el.tplAlias).toEqual(name);
    expect(el.tplASTs[name]).toEqual(el);
    expect(genData(el)).toMatchSnapshot();
  });

  it('template usage', () => {
    const is = 'foo';
    const el = {
      tag: 'template',
      attrsMap: { is },
      attrsList: [{
        name: 'is', value: is,
      }]
    };
    transformNode(el, {});

    expect(el).toMatchSnapshot();
    expect(genData(el)).toMatchSnapshot();
  });

  it('include', () => {
    const src = './__files__/template.axml';
    const el = {
      tag: 'include',
      attrsMap: { src },
      attrsList: [{
        name: 'src', value: src,
      }]
    };
    transformNode(el, { templatePath: __filename });

    expect(el).toMatchSnapshot();
    expect(genData(el)).toMatchSnapshot();
  });
});
