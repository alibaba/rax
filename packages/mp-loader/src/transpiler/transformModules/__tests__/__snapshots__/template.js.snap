// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpile module: template import 1`] = `""`;

exports[`Transpile module: template include 1`] = `
Object {
  "attrsList": Array [
    Object {
      "name": "src",
      "value": "./__files__/template.axml",
    },
  ],
  "attrsMap": Object {
    "src": "./__files__/template.axml",
  },
  "children": Array [
    Object {
      "attrsList": Array [],
      "attrsMap": Object {},
      "children": Array [],
      "parent": [Circular],
      "plain": true,
      "tag": "view",
      "type": 1,
    },
  ],
  "tag": "template",
}
`;

exports[`Transpile module: template include 2`] = `""`;

exports[`Transpile module: template template register 1`] = `""`;

exports[`Transpile module: template template usage 1`] = `
Object {
  "attrs": Array [
    Object {
      "name": "data",
      "scope": "this.data",
      "value": "{}",
    },
  ],
  "attrsList": Array [
    Object {
      "name": "is",
      "value": "foo",
    },
  ],
  "attrsMap": Object {
    "is": "foo",
  },
  "component": null,
  "tag": "$template",
}
`;

exports[`Transpile module: template template usage 2`] = `"is: _w('foo'), pageInstance: this,"`;
